packages/cashier/src/Components/Error/__tests__/withdrawal-locked.spec.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Checklist } from '@deriv/components';
+import { routes } from '@deriv/shared';
+import WithdrawalLocked from '../withdrawal-locked';
+
+jest.mock('Stores/connect', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+jest.mock('Components/Error/cashier-locked', () => () => <div>CashierLocked</div>);
+
+const setAccountStatus = (identity_status, document_status, needs_verification) => {
+    return {
+        authentication: {
+            identity: {
+                status: identity_status,
+            },
+            document: {
+                status: document_status,
+            },
+            needs_verification: [needs_verification],
+        },
+    };
+};
+
+describe('WithdrawalLocked', () => {
+    it('Should show "Check proof of identity document verification status" message and redirect to account/proof-of-identity when "-->" button clicked', () => {
+        const history = createBrowserHistory();
+        const need_poi_account_status = setAccountStatus('pending', '', '');
+
+        const wrapper = render(
+            <Router history={history}>
+                <WithdrawalLocked account_status={need_poi_account_status} is_10K_limit />
+            </Router>
+        );
+
+        const btn = wrapper.container.querySelector('.dc-checklist__item-status--action');
+        fireEvent.click(btn);
+
+        expect(history.location.pathname).toBe(routes.proof_of_identity);
+        expect(screen.getByText('Check proof of identity document verification status')).toBeInTheDocument();
+    });
+
+    it('Should show "Upload a proof of identity to verify your identity" message and redirect to account/proof-of-identity when "-->" button clicked', () => {
+        const history = createBrowserHistory();
+        const need_poi_account_status = setAccountStatus('none', '', '');
+
+        const wrapper = render(
+            <Router history={history}>
+                <WithdrawalLocked account_status={need_poi_account_status} is_10K_limit />
+            </Router>
+        );
+
+        const btn = wrapper.container.querySelector('.dc-checklist__item-status--action');
+        fireEvent.click(btn);
+
+        expect(history.location.pathname).toBe(routes.proof_of_identity);
+        expect(screen.getByText('Upload a proof of identity to verify your identity')).toBeInTheDocument();
+    });
+
+    it('Should show "Check proof of address document verification status" message and redirect to account/proof_of_address when "-->" button clicked', () => {
+        const history = createBrowserHistory();
+        const need_poa_account_status = setAccountStatus('', 'pending', 'document');
```

Comment:
We can use: `const need_poa_account_status = setAccountStatus('verified', 'pending', 'document');`